### PR TITLE
Add Day/Night Ladybug Behavior; Adds Ladybug TP move Sudden Lunge

### PR DESCRIPTION
**Pull request by CriticalXI**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

Adds mixin to control Ladybug behavior between day and night. Adds missing ladybug TP move Sudden Lunge.

## Steps to test these changes

Fight a ladybug at day or night and observe the listed changes in this PR are taking place.

### CODE REVIEW

**Comment by TeoTwawki** [comment]
image

it looks like when you start new PRs you aren't beginning on a clean copy of the base branch. You'll want to get familiar with rebasing for cleanup purposes if you aren't already.

**Comment by CriticalXI** [comment]
Unfortunately I believe this is a side effect of my fork being based off of AirSkyBoat instead of LSB, and github does not allow more than one fork for the same parent repo. Will need to explore options to fix this.

**Comment by TeoTwawki** [comment]
> Unfortunately I believe this is a side effect of my fork being based off of AirSkyBoat instead of LSB, and github does not allow more than one fork for the same parent repo. Will need to explore options to fix this.

Then the solution for your situation is to make a new branch in your fork (could call it `LSB-base`), and force-push a clean copy of LSB's base branch on top of it. Thereafter on that branch, make no changes other than to pull from LSB base branch. Whenever you start work for a new pr, base the new pr branch off that branch. Then all of them start in that clean state. Sounds more complicated than it is.

**Comment by CriticalXI** [comment]
Should be cleared up now.

File: scripts/mixins/families/ladybug.lua [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        local currentHour = VanadielHour()
+
+        if currentHour >= 18 or currentHour < 6 then
+            night(mob)
+        elseif currentHour < 18 and currentHour >= 6 then
+            day(mob)
+        end
+    end)
+
+    mob:addListener("COMBAT_TICK", "LADYBUG_COMBAT_TICK", function(mob)
```
Comment (Xaver-DaRed) [inline scripts/mixins/families/ladybug.lua]: imagen

Pretty much, you cannot name the parameter used in the listener functions "mob"

Comment (CriticalXI) [reply scripts/mixins/families/ladybug.lua]: Corrected that

**Review by Xaver-DaRed — APPROVED** [review]
LGTM

File: sql/mob_skill_lists.sql [hunk 2]
```diff
@@ -713,10 +713,10 @@ INSERT INTO `mob_skill_lists` VALUES ('Kindred',169,559);
...
 INSERT INTO `mob_skill_lists` VALUES ('Kindred',169,563);
 INSERT INTO `mob_skill_lists` VALUES ('Kindred',169,1148);
-INSERT INTO `mob_skill_lists` VALUES ('Ladybug',170,2178);
-INSERT INTO `mob_skill_lists` VALUES ('Ladybug',170,2179);
-INSERT INTO `mob_skill_lists` VALUES ('Ladybug',170,2180);
-INSERT INTO `mob_skill_lists` VALUES ('Ladybug',170,2181);
+INSERT INTO `mob_skill_lists` VALUES('Ladybug_Day', 170, 2178);
+INSERT INTO `mob_skill_lists` VALUES('Ladybug_Day', 170, 2179);
+INSERT INTO `mob_skill_lists` VALUES('Ladybug_Day', 170, 2180);
+INSERT INTO `mob_skill_lists` VALUES('Ladybug_Day', 170, 2181);
```
Comment (zach2good) [inline sql/mob_skill_lists.sql]: No need for the extra spaces in here, the rest of the file doesn't seem to use them

**Review by zach2good — CHANGES_REQUESTED** [review]
Content all seems fine. There's no need to make extra extraneous commits to fix issues, you can use `git commit --amend` (or the equivalent in your GUI of choice), and then `git push --force` to adjust your commits. When you get more advanced, you can use `git rebase`, but that probably isn't needed right now. I'd rather not squash the commits here, since I'm not sure what it does with the additional author information

**Comment by TeoTwawki** [comment]
> since I'm not sure what it does with the additional author information

It adds the person who did the rebase and push as an additional author, and retains prior authors on the resultant commits. At least that's how it works when I do it via TortoiseGit.

edit:
p.s. Tortoise also provides the opportunity to add additional persons ***during*** the rebase. On a per-commit basis.

File: sql/mob_skill_lists.sql (hunk 2 above)
Comment (CriticalXI) [reply sql/mob_skill_lists.sql]: --amend is a new one for me. I went ahead and got rid of the extra commits and amended them to the last commit. Also fixed this spacing issue.

**Review by zach2good — APPROVED** [review]

**Pull request merged**